.github/workflows/ur-source-checks.yml

```diff
@@ -0,0 +1,83 @@
+name: UR - Check generated sources
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  source_checks:
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04', 'windows-2022']
+
+    runs-on: ${{matrix.os}}
+
+    steps:
+    # TODO: If UR is merged into llvm it will require changes:
+    # - checkout only llvm repo
+    # - configure UR project from local tree
+    #
+    - name: Checkout LLVM
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Checkout UR
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: 'oneapi-src/unified-runtime'
+        path: unified-runtime
+        ref: main
+
+    - uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
+      with:
+        python-version: 3.9
+
+    - name: Install pip packages
+      working-directory: ${{github.workspace}}/unified-runtime
+      run: pip install -r third_party/requirements.txt
+
+    - name: "[Lin] Install doxygen"
+      if: matrix.os == 'ubuntu-22.04'
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y doxygen
+
+    - name: "[Win] Install doxygen"
+      if: matrix.os == 'windows-2022'
+      run: |
+        $WorkingDir = $PWD.Path
+        Invoke-WebRequest -Uri https://github.com/doxygen/doxygen/releases/download/Release_1_9_8/doxygen-1.9.8.windows.x64.bin.zip -OutFile "$WorkingDir\doxygen.zip"
+        Expand-Archive -Path "$WorkingDir\doxygen.zip"
+        Add-Content $env:GITHUB_PATH "$WorkingDir\doxygen"
+
+    - name: "[Lin] Install hwloc"
```

Comment:
imo we should split the win/lin steps into seperate workflows

Reply:
I'm willing to have them in a single file for now. The benefits for SYCL were:
1) Ability to cancel individual Linux/Windows if some CI job isn't necessary
2) Job dispatch control via "Environment" for when we had zombie processes post testing making the runner unusable for subsequent tasks.

If UR CI won't be affected by these (I'd assume it won't), then there will be little value in splitting the OSes.

Reply:
yeah my comment was mainly about confusion reading the results because people are used to how it's done for sycl, but its not a big deal

Reply:
added to the TODO list